### PR TITLE
Create intregration specs

### PR DESCRIPTION
**Pull request by Mike47ip**
#### n+1
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
    - Do not use the bullet gem in this exercise - instead analyze the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
    - **In your PR description include the screenshots of your app's logs before and after the fix**
    - This is the corresponding wireframe view: blog all user posts

#### Integration specs
- Use Capybara to write integration tests for each view in your project.
- Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.

- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

**Be sure to include integration specs for any other features you may have implemented!**

### CODE REVIEW

File: spec/features/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    it 'I can see the user first 3 posts.' do
+      expect(page).to_not have_content 'post content 1'
+    end
+    it 'I can see a button that lets me view all of a users posts.' do
+      expect(page).to have_content 'See all posts'
+    end
+    it 'When I click a users post, it redirects me to that posts show page.' do
+      visit "/user/#{@user2.id}/post/#{@post1.id}"
+      expect(page).to have_content 'post content 1'
+    end
```
Comment (omar-muhamad) [inline spec/features/user_show_spec.rb]: - [ ] This test-case is supposed to test `when I click a user's post, it redirects me to that post's show page`, not expecting the content of the page when visiting the post show page as you did. Please write a proper test for the case.

[Hint] You can do something like that:
```ruby
click_link 'post 1 title'
expect(page).to have_current_path("/users/#{@user2.id}/post/#{@post1.id}")
```

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    it 'When I click to see all posts, it redirects me to the users posts index page.' do
+      click_link 'See all posts'
+      expect(page).to have_content 'post 1 title'
+    end
```
Comment (omar-muhamad) [inline spec/features/user_show_spec.rb]: - [ ] For this case you are not testing the content of the page when the `see all posts` button is clicked. You are testing the current path you got when the `See all posts` button is clicked. Please fix that

File: spec/features/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :feature do
+  before(:each) do
+    # @user1 = User.create(name: 'Lebron', photo: 'https://lebronlinktophoto.jpg', bio: 'Make a diffrence',
+    #                      posts_counter: 0)
```
Comment (omar-muhamad) [inline spec/features/user_show_spec.rb]: - [ ] It's considered a good practice to have a readable code that is clean of any commented (unused) code. Please remove this commented code.

File: spec/features/post_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      expect(page).to have_content 'Comments: 5'
+      expect(page).to_not have_content 'Comments: 0'
+    end
+    it 'I can see how many likes a post has.' do
+      expect(page).to_not have_content 'Likes: 1'
+      expect(page).to have_content 'Likes: 0'
+    end
+    it 'When I click on a post, it redirects me to that posts show page.' do
+      click_link 'post 1 title'
+      expect(page).to have_content 'post 1 title by Lawrence'
```
Comment (omar-muhamad) [inline spec/features/post_index_spec.rb]: - [ ] Same here, You are testing the current path of the page when the post is clicked, not the content of the page. So please fix that.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi Team,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/features/user_show_spec.rb (hunk 1 above)
Comment (Mike47ip) [reply spec/features/user_show_spec.rb]: Hello @Omar-Muhamad  I have spent the last 3 days trying to implement you method but it is not working. Can we meet on zoom please?. Time is runniong out

File: spec/features/user_show_spec.rb [hunk 5]
```diff
@@ -49,13 +49,14 @@
...
+    it 'should redirect to that post show page when I click on a post' do
+      click_link 'post 2 title'
+      expect(page).to have_current_path("/user/#{@user2.id}/post/#{@post2.id}")
+      puts page.html
     end
     it 'When I click to see all posts, it redirects me to the users posts index page.' do
       click_link 'See all posts'
-      expect(page).to have_content 'post 1 title'
+      expect(page).to have_current_path("/user/#{@user2.id}/post")
     end
```
Comment (TadesseDev) [inline spec/features/user_show_spec.rb:60]: - [Optional] Dear @Mike47ip, Good job making the suggested changes, But I noticed some other of your tests are still falling. When we write tests we need to make sure all the tests are passing. Please work on these tests and make sure all are passing. 

### The issue 
Screenshot 2023-11-18 at 6 48 45 in the evening

**Review by TadesseDev — APPROVED** [review]
### Hi @Mike47ip 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**